### PR TITLE
Add tasks tests.

### PR DESCRIPTION
**Pull request by dzhuang**

### CODE REVIEW

File: course/flow.py [hunk 1]
```diff
@@ -1017,10 +1017,11 @@ def expire_flow_session(
...
 
-    if (past_due_only
-            and grading_rule.due is not None
-            and now_datetime < grading_rule.due):
-        return False
+    if past_due_only:
+        if grading_rule.due is None:
+            return False
+        elif now_datetime < grading_rule.due:
+            return False
```
Comment (dzhuang) [inline course/flow.py]: Do you think if due is None, then never due is a better behavior?

Comment (inducer) [reply course/flow.py]: I think that's what that implements, no?

Comment (dzhuang) [reply course/flow.py]: No,  when doing bulk expire in progress session.

Comment (inducer) [reply course/flow.py]: Oh, whoops. Half-asleep apparently. I looked at the code with the green background. :)

Yeah, that seems fine semantically.

Comment (dzhuang) [reply course/flow.py]: : -). NVM.

File: course/flow.py [hunk 2]
```diff
@@ -1243,10 +1244,10 @@ def finish_flow_session_standalone(
...
 
-    if grading_rule.due is not None:
-        if (
-                past_due_only
-                and now_datetime_filled < grading_rule.due):
+    if past_due_only:
+        if grading_rule.due is None:
+            return False
+        elif now_datetime_filled < grading_rule.due:
             return False
```
Comment (dzhuang) [inline course/flow.py]: Same as above.

**Comment by dzhuang** [comment]
I am using `@override_settings(CELERY_TASK_ALWAYS_EAGER=True)` to fake the async operation, that made the assumption there are no serialization issues.

**Comment by inducer** [comment]
Awesome. Thank you for working on this.

**Pull request merged**